lib/configs/filesystem.ts

```diff
@@ -22,4 +26,63 @@ export const filesystem: Linter.Config[] = [
 			'**/package-lock.json',
 		],
 	},
+
+	{
+		name: 'nextcloud/filesystem/git-ignores',
+		ignores: await getGitIgnoreContent(),
+	},
 ]
+
+/**
+ * Retrieve the content of a .gitignore file of the current project.
+ * If no file could be found then undefined is returned.
+ * Otherwise the (non-comment) entries are returned.
+ *
+ * (They have the same pattern format as ESLint ignores).
+ */
+async function getGitIgnoreContent() {
+	const file = await getGitIgnoreFile()
+	if (!file) {
+		return
+	}
+
+	const buffer = await readFile(file)
+	return buffer.toString()
+		.split('\n')
+		.map((line) => line.trim())
+		.filter((line) => line && !line.startsWith('#'))
+}
+
+/**
+ * Try to detect a .gitignore file in the current project.
+ * Either returns the path of it or undefined if nothing found.
+ */
+async function getGitIgnoreFile() {
+	let file = join(cwd(), '.gitignore')
+	if (await exists(file)) {
+		return file
+	}
+
+	const segments = import.meta.dirname.split(sep)
+	const index = segments.indexOf('node_modules')
+	if (index) {
+		file = join(segments.slice(0, index).join(sep), '.gitignore')
+		if (await exists(file)) {
+			return file
+		}
+	}
+}
```

Comment:
Could you explain the logic behind this search?

It is not just searching for `.gitignore` in the root. But also not recursive search.